File: Backend/Menus/Menu_Hostess/dashboard_hostess.js
async function cargarDashboardHostess() {
    const res = await fetch("/api/dashboard/hostess");
    const data = await res.json();

    document.getElementById("cardClientesEspera").textContent = data.clientes_espera;
    document.getElementById("cardMesasOcupadas").textContent = data.mesas_ocupadas;
    document.getElementById("cardMesasLibres").textContent = data.mesas_libres;
    document.getElementById("cardPromedioServicio").textContent = data.promedio_servicio + " min";
    document.getElementById("cardPromocionesVigentes").textContent = data.promociones_vigentes;

    const reservacionesBox = document.getElementById("listaReservacionesRecientes");
    reservacionesBox.innerHTML = "";

    data.reservaciones.forEach(r => {
        reservacionesBox.innerHTML += `
            <div class="list-item">
                <span class="list-title">${r.nombre_cliente} – ${r.no_personas} personas</span>
                <span class="list-subtitle">Hoy • ${r.hora} • ${r.estado}</span>
            </div>
        `;
    });

    const esperaBox = document.getElementById("listaEsperaResumen");
    esperaBox.innerHTML = "";

    data.espera.forEach(e => {
        esperaBox.innerHTML += `
            <div class="list-item">
                <span class="list-title">${e.nombre} – ${e.no_personas} personas</span>
                <span class="list-subtitle">En lista de espera</span>
            </div>
        `;
    });
}

document.addEventListener("DOMContentLoaded", cargarDashboardHostess);

File: Backend/Menus/Menu_Hostess/dashboard_hostess.py
from flask import Blueprint, jsonify, render_template
from db import get_connection

dashboard_hostess_bp = Blueprint("dashboard_hostess_bp", __name__)

@dashboard_hostess_bp.route("/hostess")
def hostess_dashboard():
    return render_template("hostess_dashboard.html")


@dashboard_hostess_bp.route("/api/dashboard/hostess")
def api_dashboard_hostess():
    conn = get_connection()
    cursor = conn.cursor(dictionary=True)

    cursor.execute("""
        SELECT COUNT(*) AS total
        FROM Lista_Espera
        WHERE estado = 'ESPERANDO'
    """)
    clientes_espera = cursor.fetchone()["total"]

    cursor.execute("SELECT COUNT(*) AS total FROM Mesa WHERE estado='ocupada'")
    mesas_ocupadas = cursor.fetchone()["total"]

    cursor.execute("SELECT COUNT(*) AS total FROM Mesa WHERE estado='libre'")
    mesas_libres = cursor.fetchone()["total"]

    cursor.execute("""
        SELECT AVG(TIMESTAMPDIFF(MINUTE, hora_inicio, NOW())) AS promedio
        FROM Mesa
        WHERE estado='ocupada'
          AND hora_inicio IS NOT NULL
    """)
    promedio_servicio = cursor.fetchone()["promedio"] or 0

    cursor.execute("""
        SELECT COUNT(*) AS total
        FROM Promocion
        WHERE estado = 1
          AND CURDATE() BETWEEN vigencia_inicio AND vigencia_fin
    """)
    promociones_vigentes = cursor.fetchone()["total"]

    cursor.execute("""
        SELECT nombre_cliente, no_personas, hora, estado
        FROM Reservacion
        WHERE fecha = CURDATE()
        ORDER BY hora ASC
        LIMIT 4
    """)
    reservaciones = cursor.fetchall()

    cursor.execute("""
        SELECT nombre, no_personas
        FROM Lista_Espera
        WHERE estado = 'ESPERANDO'
        ORDER BY id_lista ASC
        LIMIT 3
    """)
    espera = cursor.fetchall()

    for r in reservaciones:
        if r["hora"]:
            r["hora"] = str(r["hora"])

    cursor.close()
    conn.close()

    return jsonify({
        "clientes_espera": clientes_espera,
        "mesas_ocupadas": mesas_ocupadas,
        "mesas_libres": mesas_libres,
        "promedio_servicio": round(float(promedio_servicio), 1),
        "promociones_vigentes": promociones_vigentes,
        "reservaciones": reservaciones,
        "espera": espera
    })